### PR TITLE
fix the adapt to delegate

### PR DESCRIPTION
**Pull request by hchoura**

### CODE REVIEW

File: src/foam/dao/CopyFromDAO.js [hunk 1]
```diff
@@ -116,8 +116,16 @@ foam.CLASS({
         FObject ofObject;
         try {
           ofObject = (FObject) obj;
-          delegateObject = (FObject) getTo().newInstance();
-          delegateObject = delegateObject.copyFrom(ofObject);
+
+          if ( ofObject.isPropertySet("id") ) {
+            delegateObject = (FObject) getDelegate().find(obj.getProperty("id"));
+            delegateObject = delegateObject.copyFrom(ofObject);
+          }else {
```
Comment (kgrgreer) [inline src/foam/dao/CopyFromDAO.js:123]: Add a space after the }.

File: src/foam/dao/CopyFromDAO.js [hunk 2]
```diff
@@ -116,8 +116,16 @@ foam.CLASS({
...
+            delegateObject = (FObject) getTo().newInstance();
+            delegateObject = delegateObject.copyFrom(ofObject);
+            //TODO the id should be undefined, that it will be set automatically
+            delegateObject.setProperty("id", new Long(0));
```
Comment (kgrgreer) [inline src/foam/dao/CopyFromDAO.js:127]: Is this necessary? What if the id isn't a Long property?

Comment (hchoura) [reply src/foam/dao/CopyFromDAO.js]: We need to put the id undefined that this object will be conceder as new object record to be saved automatically.
I change it to null.

File: src/foam/dao/CopyFromDAO.js [hunk 3]
```diff
@@ -116,8 +116,16 @@ foam.CLASS({
...
-          delegateObject = delegateObject.copyFrom(ofObject);
+
+          if ( ofObject.isPropertySet("id") ) {
+            delegateObject = (FObject) getDelegate().find(obj.getProperty("id"));
+            delegateObject = delegateObject.copyFrom(ofObject);
+          } else {
+            delegateObject = (FObject) getTo().newInstance();
+            delegateObject = delegateObject.copyFrom(ofObject);
+            //TODO the id should be undefined, that it will be set automatically
+            delegateObject.setProperty("id", null);
```
Comment (hchoura) [inline src/foam/dao/CopyFromDAO.js:127]: null will create an issue

**Pull request merged**